gef-ui/api-tests/bank-review-date.get.api-test.js

```diff
@@ -0,0 +1,126 @@
+        mockIsUsingFacilityEndDate(false);
+        mockDealVersion(1);
+      });
+
+      withRoleValidationApiTests({
+        makeRequestWithHeaders: (headers) => get(`/application-details/${dealId}/facilities/${facilityId}/bank-review-date`, {}, headers),
+        whitelistedRoles: [MAKER],
+        successCode: 200,
+      });
+    });
+
+    describe('when the user is a maker', () => {
+      let sessionCookie;
+      beforeEach(async () => {
+        ({ sessionCookie } = await storage.saveUserSession([MAKER]));
+      });
+
+      it('returns 200 when the deal v1 and facility is not using facility end date', async () => {
+        // Arrange
+        mockIsUsingFacilityEndDate(false);
+        mockDealVersion(1);
+
+        // Act
+        const response = await getBankReviewDateWithSessionCookie(sessionCookie);
+
+        // Assert
+        expect(response.status).toBe(200);
+      });
+
+      it('redirects the user to the application details page if the deal is version 0', async () => {
+        // Arrange
+        mockIsUsingFacilityEndDate(false);
+        mockDealVersion(0);
+
+        // Act
+        const response = await getBankReviewDateWithSessionCookie(sessionCookie);
+
+        // Assert
+        expect(response.status).toBe(302);
+        expect(response.headers.location).toBe(`/gef/application-details/${dealId}`);
+      });
+
+      it('redirects the user to the application details page if version is 1 & isUsingFacilityEndDate is null ', async () => {
+        // Arrange
+        mockIsUsingFacilityEndDate(null);
+
+        mockDealVersion(1);
+
+        // Act
+        const response = await getBankReviewDateWithSessionCookie(sessionCookie);
+
+        // Assert
+        expect(response.status).toBe(302);
+        expect(response.headers.location).toBe(`/gef/application-details/${dealId}`);
+      });
+
+      it('redirects the user to the application details page if version is 1 & isUsingFacilityEndDate is true', async () => {
+        // Arrange
+        mockIsUsingFacilityEndDate(true);
+        mockDealVersion(1);
+
+        // Act
+        const response = await getBankReviewDateWithSessionCookie(sessionCookie);
+
+        // Assert
+        expect(response.status).toBe(302);
+        expect(response.headers.location).toBe(`/gef/application-details/${dealId}`);
+      });
+    });
+  });
+});
+
+function mockIsUsingFacilityEndDate(isUsingFacilityEndDate) {
+  when(api.getFacility).calledWith({ facilityId, userToken: expect.anything() }).mockResolvedValueOnce({ details: { isUsingFacilityEndDate } });
+}
+
+function mockDealVersion(version) {
+  when(api.getApplication).calledWith({ dealId, userToken: expect.anything() }).mockResolvedValueOnce({ version });
+}
+
+function getBankReviewDateWithSessionCookie(sessionCookie) {
+  return get(
+    `/application-details/${dealId}/facilities/${facilityId}/bank-review-date`,
+    {},
+    {
+      Cookie: [`dtfs-session=${encodeURIComponent(sessionCookie)}`],
+    },
+  );
+}
```

Comment:
Should these functions be towards the top of the file, i'd expect a "used before defined" type lint/prettier error - more for my own understanding, is this a standardised approach in GEF/DTFS?

Reply:
Ah so we've changed this on DTFS for test files only, to allow functions to be defined after use. This allows us to have the tests first (which are the important bits of a test file!) and makes the test files read clearer